src/scheduler/node_partition.c

```diff
@@ -277,6 +277,35 @@ dup_node_partition(node_partition *onp, server_info *nsinfo)
 	return nnp;
 }
 
+/**
+ * @brief copy a node partition array from pointers out of another.
+ * @param[in] onp_arr - old node partition array
+ * @param[in] new_nps - node partition array with new pointers
+ * 
+ * @return node_partition **
+ */
+node_partition **
+copy_node_partition_ptr_array(node_partition **onp_arr, node_partition **new_nps)
+{
+	int cnt;
+	int i;
+	node_partition **nnp_arr;
+
+	if (onp_arr == NULL || new_nps == NULL)
+		return NULL;
+
+	cnt = count_array((void **)onp_arr);
+	if ((nnp_arr = calloc(cnt + 1, sizeof(node_partition *))) == NULL) {
```

Comment:
I thought about it, but that would have meant I'd have to walk through the entire node search code and make sure I always added the NULL in the right place.  I was worried I'd miss some place.  In any case, calloc() didn't show up on the callgrind graph.  I don't think it is slowing things.  If you want, I can try and use malloc() and see if how hard it would be to modify the node search code.

Reply:
but you loop until 'cnt' in the loop below, so pretty much all NULL values of that array, except the last one, are being overwritten in that loop right?

Reply:
Oh, sorry, I thought you meant the calloc() in eval_selspec().  You're right, this one can be a malloc().